pages/unsubscribe/[id].tsx

```diff
@@ -1,51 +1,75 @@
 import Head from 'next/head';
 import Link from 'next/link';
 import Layout from '../../src/components/partials/Layout';
-import { decryptSignedApiKey } from '../../src/service/api-key-service';
 import { SubscriptionService } from '../../src/service/subscription-service';
 import { decrypt } from '../../src/utils/encryption';
 import { NewsletterSubscriptionService } from '../../src/service/newsletter/newsletter-subscription-service';
 import { NewsletterType } from '../../src/types/newsletter';
 import { deleteSaveSearch } from '../../src/service/saved_search_service';
 import ServiceErrorPage from '../service-error/index.page';
+import {
+  getTypeFromNotificationIds,
+  getUnsubscribeReferenceFromId,
+  removeUnsubscribeReference,
+} from '../../src/service/unsubscribe.service';
 
-export async function getServerSideProps({ query: { jwt = '' } = {} }) {
+export async function getServerSideProps({ query: { id = '' } = {} }) {
   let emailAddress: string,
-    type: keyof typeof UNSUBSCRIBE_HANDLER_MAP,
-    id: NotificationKey;
+    notificationId: NotificationKey,
+    notificationType: keyof typeof UNSUBSCRIBE_HANDLER_MAP;
   try {
-    const decodedJwt = decryptSignedApiKey(jwt);
-    type = decodedJwt.type;
-    id = decodedJwt.id;
-    emailAddress = await decrypt(decodedJwt.emailAddress);
-    await handleUnsubscribe(type, id, emailAddress);
+    const {
+      user: { encryptedEmailAddress },
+      subscriptionId,
+      newsletterId,
+      savedSearchId,
+    } = await getUnsubscribeReferenceFromId(id as string);
+
+    notificationType = getTypeFromNotificationIds({
+      subscriptionId,
+      newsletterId,
+      savedSearchId,
+    });
+    emailAddress = await decrypt(encryptedEmailAddress);
+    notificationId = subscriptionId ?? newsletterId ?? savedSearchId;
+
+    await handleUnsubscribe(notificationType, notificationId, emailAddress);
+    await removeUnsubscribeReference(id);
```

Comment:
What's the difference between these calls? Is there a reason they're separate? From their names sounds like handleUnsubscribe does not actually unsubscribe you and we need this additional separate step. If that's the case, would suggest merging the endpoints it calls into one potentially

Reply:
have combined these calls with the lastest commits